protocols/kad/src/behaviour.rs

```diff
@@ -2539,18 +2563,8 @@ pub type GetRecordResult = Result<GetRecordOk, GetRecordError>;
 /// The successful result of [`Kademlia::get_record`].
 #[derive(Debug, Clone)]
 pub struct GetRecordOk {
-    /// The records found, including the peer that returned them.
-    pub records: Vec<PeerRecord>,
-    /// If caching is enabled, these are the peers closest
-    /// _to the record key_ (not the local node) that were queried but
-    /// did not return the record, sorted by distance to the record key
-    /// from closest to farthest. How many of these are tracked is configured
-    /// by [`KademliaConfig::set_caching`]. If the lookup used a quorum of
-    /// 1, these peers will be sent the record as a means of caching.
-    /// If the lookup used a quorum > 1, you may wish to use these
-    /// candidates with [`Kademlia::put_record_to`] after selecting
-    /// one of the returned records.
-    pub cache_candidates: BTreeMap<kbucket::Distance, PeerId>,
+    /// The record found in this iteration, including the peer that returned it.
+    pub record: Option<PeerRecord>,
```

Comment:
I find it confusing that this is an `Option`. At least we need to document what `None` means here. Or we refactor `GetRecordOk`:

```rust
enum GetRecordOk {
  FoundRecord (PeerRecord),
  FinishedWithNoAdditionalRecord,
}
```

Reply:
good idea, fixed